src/pages/Login.tsx

```diff
@@ -0,0 +1,64 @@
+import React, { useState } from "react";
+import styled from "@emotion/styled/macro";
+import { useAuthDispatchContext, useAuthStateContext } from "../context/auth";
+import { Redirect, RouteComponentProps } from "react-router-dom";
+import { Location } from "history";
+
+const getReferer = (location: Location) => {
+  if (location && location.state && "referer" in location.state) {
+    const { referer } = location.state;
+    return referer;
+  }
+  return "/";
+};
+
+export const Login: React.FC<RouteComponentProps> = ({ location }) => {
+  const [username, setUsername] = useState("");
+  const [password, setPassword] = useState("");
+
+  const { login } = useAuthDispatchContext();
+  const { isAuthenticated } = useAuthStateContext();
+
+  const loginHandler = () => {
+    login({ username, password });
+  };
+
+  if (isAuthenticated) {
+    return <Redirect to={getReferer(location)} />;
+  }
+
+  return (
+    <Wrapper>
+      <label>Username</label>
+      <input
+        type="text"
+        name="username"
+        value={username}
+        onChange={e => {
+          setUsername(e.target.value);
```

Comment:
I know that putting the lambda in the `onChange` is "not performant" but this page is barebones and will be replaced once we have designs so it does not matter